engine/src/CalciteExpressionParsing.cpp

```diff
@@ -134,9 +134,19 @@ std::unique_ptr<cudf::scalar> get_scalar_from_string(const std::string & scalar_
 	if(type.id() == cudf::type_id::TIMESTAMP_DAYS) {
 		return strings::str_to_timestamp_scalar(scalar_string, type, "%Y-%m-%d");
 	}
-	if(type.id() == cudf::type_id::TIMESTAMP_SECONDS || type.id() == cudf::type_id::TIMESTAMP_MILLISECONDS
-		|| type.id() == cudf::type_id::TIMESTAMP_MICROSECONDS || type.id() == cudf::type_id::TIMESTAMP_NANOSECONDS) {
-		if (scalar_string.find(":") != std::string::npos){
+	if(type.id() == cudf::type_id::TIMESTAMP_SECONDS) {
+		if (scalar_string.find("-") != std::string::npos && scalar_string.find(":") != std::string::npos){
+			return strings::str_to_timestamp_scalar(scalar_string, type, "%Y-%m-%d %H:%M:%S");
+		} else {
+			return strings::str_to_timestamp_scalar(scalar_string, type, "%Y-%m-%d");
```

Comment:
This else assumes a specific format, but we are not checking for that format. So you should add an if here, and then in the else throw an error if its not going to be a supported format.
You could also be more ambitious and add more formats, for example "%Y/%m/%d"